src/components/tl/nccl/tl_nccl.h

```diff
@@ -13,6 +13,12 @@
 #include "components/tl/ucc_tl_log.h"
 #include "utils/ucc_mpool.h"
 
+#if CUDART_VERSION >= 11000 && (NCCL_VERSION_CODE >= NCCL_VERSION(2,10,3))
```

Comment:
@lappazos this does not compile (i guess you haven't try building with NCCL). you use macro NCCL_VERSION before including nccl.h so it is not defined at line 16. Hence compilation fails. plz remove #if at line 16 and just keep BF16.

Reply:
@vspetrov i'm keeping the if on CUDA though, right?

Reply:
i'm not sure it does any difference but you can keep CUDART_VERSION just for indication that it works for that cuda ver.